tripy/examples/segment-anything-model-v2/sam2/modeling/backbones/image_encoder.py

```diff
@@ -56,7 +56,6 @@ def __call__(self, sample: tp.Tensor):
         # Forward through backbone
         if self.compiled_executable:
             features_pos = list(self.compiled_executable(sample))
-            tp.default_stream().synchronize()
```

Comment:
Why do you  think this is not required here? Are you synchronizing in another location?

We currently don't do anything in our `__dlpack__` method to ensure that the tensors are ready for the external consumer. We should be synchronizing there or inserting the proper CUDA event-based sychronizations to sychronize the MLIR-TRT runtime stream (which I'm guess what `tp.default_stream()`  points to) with the Torch default stream.

Reply:
Good point. I've updated the demo to use the Tripy default stream for PyTorch operations as well. The synchronization happens implicitly when we call `(out_mask_logits[i] > 0.0).cpu()` at the end of each loop iteration.